smac/facade/hydra_facade.py

```diff
@@ -0,0 +1,144 @@
+from smac.tae.execute_ta_run_hydra import ExecuteTARun
+from smac.tae.execute_ta_run_hydra import ExecuteTARunAClib
+from smac.scenario.scenario import Scenario
+from smac.facade.smac_facade import SMAC
+from smac.utils.io.output_directory import create_output_directory
+from smac.runhistory.runhistory import RunHistory
+from smac.runhistory.runhistory import DataOrigin
+from smac.optimizer.objective import average_cost
+
+__author__ = "Marius Lindauer"
+__copyright__ = "Copyright 2017, ML4AAD"
+__license__ = "3-clause BSD"
+
+
+class Hydra(object):
+    """Facade to use Hydra default mode
+
+    Attributes
+    ----------
+    logger
+    stats : Stats
+        loggs information about used resources
+    solver : SMBO
+        handles the actual algorithm calls
+    runhistory : RunHistory
+        List with information about previous runs
+    trajectory : list
+        List of all incumbents
+    """
+
+    def __init__(self,
+                 scenario: Scenario,
+                 n_iterations: int,
+                 run_id: int=1,
+                 tae: typing.Type[ExecuteTARun]=ExecuteTARunOld,
+                 **kwargs):
+        """Constructor
+
+        Parameters
+        ----------
+        scenario : ~smac.scenario.scenario.Scenario
+            Scenario object
+        n_iterations: int,
+            number of Hydra iterations
+        """
+
+        self.logger = logging.getLogger(
+            self.__module__ + "." + self.__class__.__name__)
+
+        self.n_iterations = n_iterations
+
+        self.logger = logging.getLogger(
+            self.__module__ + "." + self.__class__.__name__)
+
+        self.scenario = scenario
+        self.run_id = run_id
+        self.kwargs = kwargs
+        self.output_dir = None
+        self.top_dir = None
+        self.solver = None
+        self.rh = RunHistory(average_cost)
+        self.tae = tae(ta=self.scenario.ta, run_obj=self.scenario.run_obj)
+        self.tae_type = tae
+
+    def optimize(self):
+        """Optimizes the algorithm provided in scenario (given in constructor)
+
+        Returns
+        -------
+        portfolio : typing.List[Configuration]
+            Portfolio of found configurations
+        """
+
+        portfolio = []
+        portfolio_cost = np.inf
+        if self.output_dir is None:
+            self.top_dir = "hydra-output_%s" % (
+                datetime.datetime.fromtimestamp(time.time()).strftime('%Y-%m-%d_%H:%M:%S_%f'))
+            self.scenario.output_dir = os.path.join(self.top_dir, "smac3-output_%s" % (
+                datetime.datetime.fromtimestamp(time.time()).strftime('%Y-%m-%d_%H:%M:%S_%f')))
+            self.output_dir = create_output_directory(self.scenario, run_id=self.run_id, logger=self.logger)
+
+        self.solver = SMAC(scenario=self.scenario, tae_runner=self.tae, **self.kwargs)
+        for i in range(self.n_iterations):
+            self.logger.info("Iteration: %d", (i + 1))
```

Comment:
Please change to "Hydra iteration:"

Reply:
And maybe add a special line before. e.g
self.logger.info("##########################################################")